File: scripts/check_images.py
import re

with open('src/data/mockData.ts', 'r', encoding='utf-8') as f:
    content = f.read()

urls = re.findall(r'(https://images\.unsplash\.com/[^\s"\'<>]+)', content)
clean_urls = set()
for u in urls:
    clean_urls.add(u.split('?')[0])

print(f"Total URL occurrences: {len(urls)}")
print(f"Unique base URLs: {len(clean_urls)}")

File: scripts/check_images.js
const fs = require('fs');
const content = fs.readFileSync('src/data/mockData.ts', 'utf8');
const units = JSON.parse(content.replace('export const mockApartmentUnits: ApartmentUnit[] = ', '').replace(/;\s*$/, ''));

console.log('Total units:', units.length);
const firstImages = units.map(u => u.images[0]);
const uniqueFirstImages = new Set(firstImages);
console.log('Unique first images:', uniqueFirstImages.size, 'out of', units.length);

const allImages = units.flatMap(u => u.images);
console.log('Total images:', allImages.length, 'Unique total images:', new Set(allImages).size);
